src/main/java/Duke.java

```diff
@@ -1,10 +1,226 @@
+            // relative path is different for testing
+            System.out.println("Check file path!");
+        }
+    }
+
+    public static void writeTasksToFile() throws IOException {
+        FileWriter file = new FileWriter(DIRECTORY_NAME + "/" + FILE_NAME);
+        for (Task t : tasks) {
+            file.write(t.getCode() + "|" + (t.isDone() ? "1" : "0") + "|" +
+                    t.getDescription() + "|" + t.getDatetime() + System.lineSeparator());
+        }
+        file.close();
+    }
+
+    public static void loadTasks() {
+        File directory = new File(DIRECTORY_NAME);
+        boolean directoryCreated = false;
+        try {
+            if (!directory.exists()) {
+                directoryCreated = directory.mkdir();
+            }
+        } catch (Exception e) {
+            e.getStackTrace();
+        }
+        // created = True -> first time running -> no need to load
+        // created = False -> directory exists OR fail to create (exists)
+        if (!directoryCreated) {
+            // handle directory exists case
+            try {
+                readTasksFromFile();
+            } catch (FileNotFoundException e) {
+                // relative path is different for testing
+                System.out.println("File not found!");
+            }
+        }
+    }
+
+    public static void readTasksFromFile() throws FileNotFoundException {
+        File file = new File(DIRECTORY_NAME + "/" + FILE_NAME);
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNext()) {
+            processLine(scanner.nextLine());
+        }
+    }
+
+    public static void processLine(String line) {
+        String[] parts = line.split("\\|");
+        boolean done = (!"0".equals(parts[1]));
+        switch (parts[0]) {
+        case "T":
+            tasks.add(new Todo(done, parts[2]));
+            break;
+        case "E":
+            tasks.add(new Event(done, parts[2], parts[3]));
+            break;
+        case "D":
+            tasks.add(new Deadline(done, parts[2], parts[3]));
+            break;
+        default:
+            break;
+        }
+    }
+
+    public static boolean executeCommand(String rawCommand) {
+        boolean loop = true;
+        // remove leading and trailing space
+        String trimmedCommand = rawCommand.trim();
+        try {
+            if (trimmedCommand.startsWith("bye")) {
+                bye();
+                loop = false;
+            } else if (trimmedCommand.startsWith("list")) {
+                list();
+            } else if (trimmedCommand.startsWith("done")) {
+                int taskNum = Integer.parseInt(trimmedCommand.split(" ")[1]);
+                markAsDone(taskNum);
+            } else if (trimmedCommand.startsWith("delete")) {
+                int taskNum = Integer.parseInt(trimmedCommand.split(" ")[1]);
+                deleteTask(taskNum);
+            } else {
+                addTask(trimmedCommand);
+            }
```

Comment:
The fall-through else seems odd here.
Because you'd be assign all commands that doesn't start with any of the above to be `addTask`
I deduce you may be using it just to throw an UnknownCommandException, but wouldn't it be better to throw one here?
```suggestion
            } else if (trimmedCommand.startsWith("add")){
                addTask(trimmedCommand);
            } else {
                throw new UnknownCommandException()
            }
```

It seems out of place to dedicate the handling of unknown command to the add command

Reply:
I think i will fix this later when i try to abstract out Command class. Thanks for the comment.